pyaedt/modeler/stackup_3d.py

```diff
@@ -0,0 +1,2284 @@
+from collections import OrderedDict
+
+from pyaedt.generic.general_methods import generate_unique_name
+from pyaedt.generic.general_methods import pyaedt_function_handler
+from pyaedt.modules.MaterialLib import Material
+
+LAYERS = {"s": "signal", "g": "ground", "d": "dielectric"}
+
+
+def _replace_by_underscore(character, string):
+    if not isinstance(character, str):
+        raise TypeError("character must be str")
+    if not isinstance(character, str):
+        raise TypeError("string must be str")
+    reformat_name = list(string)
+    while character in reformat_name:
+        index = reformat_name.index(character)
+        reformat_name[index] = "_"
+    return "".join(reformat_name)
+
+
+class NamedVariable(object):
+    """Cast Pyaedt Variable object to simplify getter and setters in Stackup3D.
+
+    The returned Polyline object exposes the methods for manipulating the polyline.
+
+    Parameters
+    ----------
+    application : :class:`pyaedt.hfss.Hfss
+        The Hfss design or project where the variable will be create
+    name : str
+        The name of the variable. If the the name begins by a '$', the variable will be a project variable,
+        else it will be a design variable.
```

Comment:
```suggestion
        The name of the variable. If the the name begins with an '$', the variable will be a project variable.
        Otherwise, it will be a design variable.
```